user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -4,21 +4,59 @@
  */
 package io.strimzi.operator.user;
 
-import io.strimzi.api.kafka.model.CertificateAuthority;
 import io.strimzi.operator.common.InvalidConfigurationException;
 import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.operator.resource.AbstractConfig;
+
 
 import java.io.IOException;
 import java.io.StringReader;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Collections;
+
+import static io.strimzi.operator.common.operator.resource.AbstractConfig.BOOLEAN;
+import static io.strimzi.operator.common.operator.resource.AbstractConfig.LABEL_PREDICATE;
+import static io.strimzi.operator.common.operator.resource.AbstractConfig.STRING;
+import static io.strimzi.operator.common.operator.resource.AbstractConfig.INTEGER;
+import static io.strimzi.operator.common.operator.resource.AbstractConfig.LONG;
 
 /**
  * Cluster Operator configuration
  */
 public class UserOperatorConfig {
+
+    private static final Map<String, ConfigParameter<?>> CONFIG_VALUES = new HashMap<>();
+
+    /**
+     * Models a configuration parameter, identified by a unique key, which may be required, and if not may have a default value.
+     * Optional parameters without a default value implicitly have a null default.
+     * The key is also the name of the environment variable from which the value may be read, when it is read from the environment.
+     *
+     * @param key Configuration parameter name/key
+     * @param <T> Type of object
+     * @param type type of the default value
+     * @param defaultValue default value of the configuration parameter
+     * @param required  If the value is required or not
+     */
+    public record ConfigParameter<T>(String key, AbstractConfig<? extends T> type, String defaultValue, boolean required) {
+        /**
+         * Contructor
+         * @param key Configuration parameter name/key
+         * @param type type of the default value
+         * @param defaultValue default value of the configuration parameter
+         * @param required  If the value is required or not
+         */
+        public ConfigParameter {
+            CONFIG_VALUES.put(key, this);
+        }
+    }
```

Comment:
Good catch jakub. I think I can use something like this in CO also

Reply:
@scholzj I tried to move this method to `operator-common` but there is one problem. Iam currently using the contructor to build the Configuration Map as a side effect of initialising the parameters. Now If I move it to `operator-common` it would be hard to do it the way I am doing it currently

Reply:
That is fine. If it cannot be done it cannot be done.